### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by gautamxyz**
Completed practice and demo

### CODE REVIEW

File: experiment-descriptor.json [hunk 1]
```diff
@@ -34,13 +34,27 @@
...
     },
+    {
+      "target": "demo-pass-transistor.html",
+      "source": "simulation/demo-pass-transistor.html",
+      "label": "Demo",
+      "unit-type": "task",
+      "content-type": "simulation"
+    },
     {
       "target": "simulation.html",
```
Comment (raj-vlabs) [inline experiment-descriptor.json:45]: Can we call this file practice.html?

File: experiment/posttest.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            "b": "ClockComplement",
+            "c": "DataLine1",
+            "d": "DataLine2"
+        },
+        "correctAnswer": "b"
+    },
+    {
+        "question": "4. Which are following statements are true?",
+        "answers": {
+            "a": "Latches are used as temporary buffers  ",
```
Comment (raj-vlabs) [inline experiment/posttest.json:35]: Please remove extra spaces at the end of the answer

File: experiment/pretest.json [hunk 3]
```diff
@@ -1,46 +1,52 @@
...
-            "d":"None"
-          },
-"correctAnswer":"b"}            
-]
+  {
+    "question": "1. Multiplex is a ?",
+    "answers": {
+      "a": "Combinational Circuit ",
+      "b": "Combinational Circuit ",
+      "c": " Sequential Circuit",
```
Comment (raj-vlabs) [inline experiment/pretest.json:7]: Please remove extra space at the start of the option

File: experiment/pretest.json [hunk 4]
```diff
@@ -1,46 +1,52 @@
...
-"answers":{
-            "a":"Counter",
-            "b":"Data Select ",
-            "c":"Single Input Many Output",
-            "d":"None"
-          },
-"correctAnswer":"b"}            
-]
+  {
+    "question": "1. Multiplex is a ?",
```
Comment (raj-vlabs) [inline experiment/pretest.json:3]: Should it be Multiplex or multiplexer?

File: experiment/pretest.json [hunk 5]
```diff
@@ -1,46 +1,52 @@
...
+      "d": "None of the above"
+    },
+    "correctAnswer": "c"
+  },
+  {
+    "question": "3. Transmission gate is preferred over Complementary CMOS logic because ____",
+    "answers": {
+      "a": "Transistor count required to implement a logic function in transmission gate is low",
+      "b": "Low Delay",
+      "c": " Both of them",
```
Comment (raj-vlabs) [inline experiment/pretest.json:27]: Please remove leading space

File: experiment/pretest.json [hunk 6]
```diff
@@ -1,46 +1,52 @@
...
+      "d": "None of the above"
+    },
+    "correctAnswer": "c"
+  },
+  {
+    "question": "4. How many 2x1 multiplexer will be require to design 8x1 multiplexer?",
+    "answers": {
+      "a": "Three ",
```
Comment (raj-vlabs) [inline experiment/pretest.json:35]: Please remove trailing space

File: experiment/simulation/demo-mux.html [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+            </select>
+            <button class="v-button" id="play-or-pause" onclick="simulationStatus()">Start</button>
+            <button class="v-button" id="restart" onclick="restartCircuit()">Reset</button>
+        </div>
+    </div>
+
+    <div class="columns is-centered is-flex-direction-row is-flex-wrap-wrap is-align-content-start">
+        <div class="column is-7-desktop is-12-tablet is-12-mobile">
+            <div id="circuit-board">
+                <svg class="svg" xmlns="http://www.w3.org/2000/svg" width="100%" height="120%" viewBox="0 0 925 1000 ">
```
Comment (raj-vlabs) [inline experiment/simulation/demo-mux.html:75]: Please use https links exclusively

File: experiment/simulation/js/mux.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        changeObservation("&#10060; Circuit is incorrect", 'text-success', 'text-danger');
+    }
+}
+
+export function ptValidate() {
+    if ((connectionMap.has("input0$pmos0") && connectionMap.has("clock0$pmos0") && connectionMap.has("clockbar0$nmos0") && connectionMap.has("pmos0$output0") && connectionMap.has("input0$nmos0") && connectionMap.has("nmos0$output0") && connectionMap.size === 6 || (connectionMap.has("input1$pmos0") && connectionMap.has("clock0$pmos0") && connectionMap.has("clockbar0$nmos0") && connectionMap.has("pmos0$output0") && connectionMap.has("input1$nmos0") && connectionMap.has("nmos0$output0") && connectionMap.size === 6)) || (connectionMap.has("input0$pmos0") && connectionMap.has("clockbar0$pmos0") && connectionMap.has("clock0$nmos0") && connectionMap.has("pmos0$output0") && connectionMap.has("input0$nmos0") && connectionMap.has("nmos0$output0") && connectionMap.size === 6 || (connectionMap.has("input1$pmos0") && connectionMap.has("clockbar0$pmos0") && connectionMap.has("clock0$nmos0") && connectionMap.has("pmos0$output0") && connectionMap.has("input1$nmos0") && connectionMap.has("nmos0$output0") && connectionMap.size === 6))) {
```
Comment (raj-vlabs) [inline experiment/simulation/js/mux.js:55]: Please consider moving this huge condition to a separate function

**Review by raj-vlabs — CHANGES_REQUESTED** [review]
1. Please replace all http references with https

File: experiment-descriptor.json (hunk 1 above)
Comment (gautamxyz) [reply experiment-descriptor.json:45]: changed to practice.html

File: experiment/simulation/demo-mux.html (hunk 7 above)
Comment (gautamxyz) [reply experiment/simulation/demo-mux.html:75]: Changed to https. However in animation.js files switching to https from http was giving errors. So we have not made changes in those files.

File: experiment/pretest.json (hunk 5 above)
Comment (gautamxyz) [reply experiment/pretest.json:27]: fixed

File: experiment/pretest.json (hunk 4 above)
Comment (gautamxyz) [reply experiment/pretest.json:3]: Corrected to multiplexer

File: experiment/pretest.json (hunk 3 above)
Comment (gautamxyz) [reply experiment/pretest.json:7]: Removed

File: experiment/posttest.json (hunk 2 above)
Comment (gautamxyz) [reply experiment/posttest.json:35]: Removed

File: experiment/simulation/js/mux.js (hunk 8 above)
Comment (gautamxyz) [reply experiment/simulation/js/mux.js:55]: done

File: experiment/posttest.json [hunk 9]
```diff
@@ -32,19 +32,19 @@
     {
         "question": "4. Which are following statements are true?",
```
Comment (raj-vlabs) [inline experiment/posttest.json:33]: Which of the following

**Review by raj-vlabs — CHANGES_REQUESTED** [review]
Code looks good. There's one grammatical mistake in a question, please fix it, I will approve after that.

File: experiment/posttest.json (hunk 9 above)
Comment (gautamxyz) [reply experiment/posttest.json]: fixed

**Review by raj-vlabs — APPROVED** [review]

**Pull request merged**